src/main/java/org/flowsoft/flowg/visitors/CodeGeneratingVisitor.java

```diff
@@ -556,17 +570,110 @@ public ExpressionValue Visit(FunctionCallNode functionCallNode) throws Exception
+                                    var = point.GetX().toPlainString();
+                                    break;
+                                }
+                                case "y" -> {
+                                    var = point.GetY().toPlainString();
+                                    break;
+                                }
+                                case "z" ->{
+                                    var = point.GetZ().toPlainString();
+                                    break;
+                                }
+                                default -> throw new WrongPointIndexingException(pointIdentifier, functionEntry.GetGCode().GetLeft(), functionEntry.GetGCode().GetRight());
+                            }
+                        }
+                        default -> throw new ExpectedTypeException(Type.Point, identifierType, functionEntry.GetGCode().GetLeft(), functionEntry.GetGCode().GetRight());
+                    }
+                }else{
+                    Type identifierType;
+                    try{
+                        identifierType = _symbolTable.LookupVariable(Identifier).GetType();
+                    }catch (Exception e){
+                        throw new SymbolNotFoundException(Identifier, functionEntry.GetGCode().GetLeft(), functionEntry.GetGCode().GetRight());
+                    }
+                    switch (identifierType){
+                        case Number -> {
+                            var = _symbolTable.LookupVariable(Identifier).GetNumber().toPlainString();
+                            break;
+                        }
+                        case Point -> {
+                            var = _symbolTable.LookupVariable(Identifier).GetPoint().toString();
+                            break;
+                        }
+                        case Boolean -> {
+                            var = _symbolTable.LookupVariable(Identifier).GetBoolean().toString();
+                            break;
+                        }
+                        case Void -> throw new ExpectedTypeException(Type.Number, Type.Void, functionEntry.GetGCode().GetLeft(), functionEntry.GetGCode().GetRight()) {
+                            //Technically not a correct error message
+                        };
+                    }
+                }
+                gCodeBody = preString + var + postString;
+                matcher = patternIdentifier.matcher(gCodeBody);
+            }
+            _symbolTable = oldSymbolTable;
+            String[] array = gCodeBody.split("\\n");
+            String gCode = "";
+            for  (String string : array){
+                String trimmedCode = string.trim();
+                if (trimmedCode != ""){
+                    gCode += string.trim() + "\n";
+                }
+
+            }
 
-        return value;
+            _stringBuilder.append(gCode);
+            return new ExpressionValue(Type.Void);
+        }
```

Comment:
This is type checking inside code generation. It is fine as long as this is replaced with jflex and cup in the future